### PR TITLE
Forbid queries endAt an uncommitted server timestamp.

### PR DESCRIPTION
**Pull request by rsgowman**
Without this, it still fails, but:
a) not until serializing the query, and
b) the error is an internal error, and
c) the error message is quite cryptic and has nothing to do with the problem.

Port of https://github.com/firebase/firebase-android-sdk/pull/138

### CODE REVIEW

File: Firestore/Example/Tests/Integration/API/FIRValidationTests.mm [hunk 1]
```diff
@@ -420,6 +420,55 @@ - (void)testQueryCannotBeCreatedFromDocumentsMissingSortValues {
...
+  XCTestExpectation *onlineCallbackDone = [self expectationWithDescription:@"online callback done"];
+
+  [collection addSnapshotListener:^(FIRQuerySnapshot *snapshot, NSError *error) {
+    XCTAssertNil(error);
+
+    // Skip the initial empty snapshot.
+    if (snapshot.empty) return;
+
+    XCTAssertEqual(snapshot.count, 1);
+    FIRQueryDocumentSnapshot *docSnap = [snapshot documents][0];
```
Comment (mikelehen) [inline Firestore/Example/Tests/Integration/API/FIRValidationTests.mm]: nit: Given these are properties, I'd do `snapshot.documents[0]` and `snapshot.metadata.pendingWrites`

File: Firestore/Source/API/FIRQuery.mm [hunk 2]
```diff
@@ -572,7 +575,15 @@ - (FSTBound *)boundFromSnapshot:(FIRDocumentSnapshot *)snapshot isBefore:(BOOL)i
                                                    databaseID:self.firestore.databaseID]];
     } else {
       FSTFieldValue *value = [document fieldForPath:sortOrder.field];
-      if (value != nil) {
+
+      if ([value isKindOfClass:[FSTServerTimestampValue class]]) {
+        FSTThrowInvalidUsage(@"InvalidQueryException",
+                             @"Invalid query. Your are trying to start or end a query using a "
```
Comment (mikelehen) [inline Firestore/Source/API/FIRQuery.mm]: "Your are" ?

File: Firestore/Example/Tests/Integration/API/FIRValidationTests.mm (hunk 1 above)
Comment (rsgowman) [reply Firestore/Example/Tests/Integration/API/FIRValidationTests.mm]: Done.

File: Firestore/Source/API/FIRQuery.mm (hunk 2 above)
Comment (rsgowman) [reply Firestore/Source/API/FIRQuery.mm]: Done.

**Pull request merged**